package/gluon-hoodselector/luasrc/usr/lib/lua/hoodselector/util.lua

```diff
@@ -0,0 +1,83 @@
+local unistd = require('posix.unistd')
+local util = require 'gluon.util'
+local math_polygon = require('math-polygon')
+local json = require 'jsonc'
+local uci = require('simple-uci').cursor()
+local site = require 'gluon.site'
+local logger = require('posix.syslog')
+local M = {}
+
+function M.log(msg)
+	io.stdout:write(msg..'\n')
+	logger.openlog(msg, logger.LOG_PID)
+end
+
+function M.get_domains()
+	local list = {}
+	for _, domain_path in ipairs(util.glob('/lib/gluon/domains/*.json')) do
+		table.insert(list, {
+			domain_code = domain_path:match('([^/]+)%.json$'),
+			domain = assert(json.load(domain_path)),
+		})
+	end
+	return list
+end
+
+-- Return the default domain from the domain list.
+-- This method can return the following data:
+-- * default domain
+-- * nil if no default domain has been defined
```

Comment:
What is the implication of `nil` here? What if there is no default domain? Will the router be soft-bricked?

Reply:
The default domain is mandatory, so the error handling here is probably unnecessary, no?

Reply:
yes the error handling is not necessary. It wars partly from the old code.